tools/migrate/main.go

```diff
@@ -309,14 +305,24 @@ func migrate(srcDir, dstDir string) error {
 		return err
 	}
 
-	hash, n, err := hashManifest(signedManifests[v1manifest.ManifestTypeSnapshot])
+	// init snapshot
+	snapshot, err = snapshot.SetVersions(signedManifests)
+	if err != nil {
+		return err
+	}
+	signedManifests[v1manifest.ManifestTypeSnapshot], err = v1manifest.SignManifest(snapshot, keys[v1manifest.ManifestTypeSnapshot]...)
+	if err != nil {
+		return err
+	}
+
+	hash, _, err := hashManifest(signedManifests[v1manifest.ManifestTypeSnapshot])
 	if err != nil {
 		return errors.Trace(err)
 	}
 	timestamp.Meta = map[string]v1manifest.FileHash{
 		v1manifest.ManifestFilenameSnapshot: {
 			Hashes: hash,
-			Length: n,
+			Length: limitLength,
```

Comment:
Is it correct? I think the length should be the real file length of snapshot.json.

Reply:
>= the length of snapshot.sjon is fine.
even zero should be ok which means no limit.

Reply:
0 is not ok (might permit DOS-style attacks), but the same or greater (within reason) is ok.

Reply:
when 0, client should use a limit size at the client side to protest this attack, then I think it's ok?
seems I read this in the TUF spec before, but I am not sure.

Reply:
I think it is true, but we have not implemented this